### PR TITLE
No string raises

### PR DESCRIPTION
**Pull request by mrshirts**
Eliminate cases where strings are raised as exceptions.

### CODE REVIEW

File: pymbar/confidenceintervals.py [hunk 1]
```diff
@@ -223,7 +223,7 @@ def generateConfidenceIntervals(replicates, K):
                     print(replicate['error'])
                     print("destimated")
                     print(replicate['destimated'])
-                    raise "isnan"
+                    raise DataError("isnan")
```
Comment (Lnaden) [inline pymbar/confidenceintervals.py]: This module does not seem to import DataError from anywhere.

**Comment by mrshirts** [comment]
Ah, missed that. Interesting that it passed in 3.x.  I'd like to not make this depend on the rest of pymbar, since its more general and doesn't depend currently.  I'm looking through https://docs.python.org/3/library/exceptions.html#Exception to see if there's a good way to do this but have to come back to this later today.

**Comment by Lnaden** [comment]
I don't think there is any test which touches that part of the code, so there is nothing to test. The only place that function is invoked is in `harmonic-oscillators-distribution.py`, which itself is never invoked anywhere from what I can tell.

**Comment by mrshirts** [comment]
Not sure how it broke this time when the appveyor ones passed.

**Comment by mrshirts** [comment]
OK, since the tests pass, and I've address the comments, I am going to merge myself.  I THINK the tests failed because it uses a random sample, and it's possible for that to fail. Should consider using a fixed seed (now possible with latest check in).

**Pull request merged**